### ISSUE TITLE
Pages are reloaded when switching between tabs

### ISSUE THREAD

**Issue by joemasilotti** [hotwired/hotwire-native-ios#40]
When switching between tabs a web view is *always* reloaded via a `.restore` visit.

[`Session` line 260](https://github.com/hotwired/hotwire-native-ios/blob/3ba9441959ead175e7b5220abbaf8f18a79bff72/Source/Turbo/Session/Session.swift#L260), introduced in [turbo-ios PR #189](https://github.com/hotwired/turbo-ios/pull/189), assumes a single navigation controller which then triggers the restoration visit. This occurs from `HotwireWebViewController.viewWillAppear(_:)` being called before the web view appears when switching between tabs.

https://github.com/user-attachments/assets/73027891-fc22-4c31-8a85-5cfb48c82c2f

Replace "switching tabs" with any native to web interaction - I'm only using tabs here because that's where I noticed the issue. But this would also occur if you had a split view controller or similar - anytime `viewWillAppear(_:)` would be called on the view controller.

### To reproduce

1. Create a new Xcode project
2. Add the Hotwire Native iOS dependency
3. Replace `SceneDelegate.swift` with code below
4. Set a breakpoint on `Session` line 260
5. Switch to the native tab then back to the web tab
6. Observe that the break point is hit and the page is reloaded

```swift
import HotwireNative
import UIKit

let baseURL = URL(string: "https://hotwire-native-demo.dev")!

class SceneDelegate: UIResponder, UIWindowSceneDelegate {
    var window: UIWindow?

    private let navigator = Navigator()

    func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
        let tabBarController = UITabBarController()
        let secondViewController = SecondViewController()
        secondViewController.title = "Second"

        tabBarController.viewControllers = [
            navigator.rootViewController,
            secondViewController
        ]
        window?.rootViewController = tabBarController

        navigator.route(baseURL)
    }
}

class SecondViewController: UIViewController {
    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = .systemGreen
    }
}
```

**Comment by karlentwistle** [comment]
Thanks for posting this @joemasilotti. I recently upgraded the Turbo Native app at my company from turbo-ios to hotwire-native-ios, which also uses tabs, and I'm encountering this exact issue.

**Comment by joemasilotti** [comment]
Closed via #50.
